modules/tags/modules.php

```diff
@@ -103,13 +103,13 @@ class Hm_Handler_imap_tag_content extends Hm_Handler_Module {
     public function process() {
         $data_sources = imap_data_sources('');
         $ids = array_map(function($ds) { return $ds['id']; }, $data_sources);
-        $tag_id = $this->request->post['folder'];
+        $tag_id = $this->request->post['tag_id'];
         if ($ids && $tag_id) {
             $limit = $this->user_config->get('tag_per_source_setting', DEFAULT_TAGS_PER_SOURCE);
             $date = process_since_argument($this->user_config->get('tag_since_setting', DEFAULT_TAGS_SINCE));
             $folder = bin2hex('INBOX');
             if (array_key_exists('folder', $this->request->post)) {
-                $folder = $this->request->post['folder'];
+                $folder = bin2hex($this->request->post['folder']);
             }
             list($status, $msg_list) = merge_imap_search_results($ids, 'ALL', $this->session, $this->cache, array(hex2bin($folder)), $limit, array(array('SINCE', $date), array('HEADER X-Cypht-Tags', $tag_id)));
```

Comment:
@Shadow243 Please enlighten me, why is the INBOX folder needed here?

Reply:
We need to send a folder, and by default it is INBOX, however in this change here https://github.com/cypht-org/cypht/blob/master/modules/core/js_modules/Hm_MessagesStore.js#L167 you send tag_id as folder but it is an error, folders are for example: INBOX, Sent, Draft, etc...

I tried to remove the folder parameter but what if we need it to filter in another folder? So I left INBOX hoping that in the future we will go further to other folders

Note: t`ag_id` is different from a `folder`

Reply:
As no explicit folder is needed here, we should keep the content of Hm_MessagesStore unchanged, that way we get an API that works for almost all lists of messages without varying properties, thus, assigning the `tag_id` to the folder property is the better way to go.

Also, does this fix the retrieval of the message rows?

Reply:
We can't assign tag_id into folder. Like I said tag_id is different of folder.

feel free to push something on this PR if you have an idea.

Reply:
Maybe you don't want to understand my point and prefer to stick to your opinion, but in the context where `tag_id` is being used, it wouldn't be unacceptable to assign its value to a property named `folder`.

Reply:
This is exactly what I will understand, if I send tag_id in folder, this line will no longer have any meaning:
```
if (array_key_exists('folder', $this->request->post)) {
                $folder = $this->request->post['folder'];
```
remember in `$this->request->post['folder']` we have the tag_id, what's your thoughts ?

Reply:
We can remove that statement because we don't use it.

Reply:
Well noted, But if we remove the folder parameter, we will no longer have a filter, we will only search in INBOX and if I have an email from the Sent folder then it will never be displayed, but by passing folder separately from the tag_id we will add the possibility of filtering, ideally this PR should rather be modified and add to the user a possibility of sending the folder filter (INBOX, Draft, Sent, etc...)

Reply:
Here's how the query worked before:
<img width="1196" alt="Screenshot 2024-11-16 at 14 23 38" src="https://github.com/user-attachments/assets/75eb04e7-1856-42a2-a226-e59a666a6f94">

Reply:
A tag is a filter by itself; implementing additional folder filters on top of it creates unnecessary clutter in this case. It's like saying, 'I want the list of messages from `Everything`, but then show me only messages from a specific folder.' I'm fairly certain the client doesn't use such filtering.

Reply:
Here is the answer hidden in what you just wrote, we can send 'Everything' this filter already exists. in merge_imap_search_results when it is the tag we must return all "INBOX,Sent, Drafts, etc...

Reply:
Let me rephrase to help you understand what I said:

Imagine someone requests a list of messages (e.g., Archive), and on top of that filter, they add, _Please show me only messages from the folder `Sent`._ Would that add any value?

Reply:
I understood your point of view and it even reminds me that I had not taken into account the fact that we must display everything (from the whole folders once it is tagged), I have just put the PR in draft because I am in the process of broadening the reflection, I suggest that you let me finish modifying then I will notify you

Reply:
@jacob-js Here is the result I was looking for, I display the all email that have been tagged from folders all folders using `get_special_use_mailboxes()`. In this screenshort the first email is in "Sent" and the rest in 'INBOX'.
<img width="1108" alt="Screenshot 2024-11-16 at 16 56 27" src="https://github.com/user-attachments/assets/ff1fd51e-935a-45c9-803b-91854e4a102c">